crates/subspace-farmer/src/single_disk_farm/farming/rayon_files.rs

```diff
@@ -0,0 +1,53 @@
+use std::fs::{File, OpenOptions};
+use std::io;
+use std::path::Path;
+use subspace_farmer_components::file_ext::{FileExt, OpenOptionsExt};
+use subspace_farmer_components::ReadAtSync;
+
+/// Wrapper data structure for multiple files to be used with [`rayon`] thread pool, where the same
+/// file is opened multiple times, once for each thread.
+pub struct RayonFiles {
+    files: Vec<File>,
```

Comment:
IIUC, the idea is to not share the files (in read mode) across threads, even if the file needs to be opened multiple times. Just curious, why not use [std](https://doc.rust-lang.org/std/index.html)::[thread_local](https://doc.rust-lang.org/std/macro.thread_local.html#) or equivalent?

Reply:
What would be the benefit of using `std::thread_local`?

Reply:
Just to avoid creating our own wrapper, but nothing major

Reply:
The wrapper is needed either way, we need to send "something" that implements `ReadAtSync` trait to sync auditing implementation and we want it to use one file handle per rayon thread, this implementation does exactly that.